tools/toolchains/arm.py

```diff
@@ -56,6 +56,10 @@ def __init__(self, target, options=None, notify=None, macros=None, silent=False,
         else:
             cpu = target.core
 
+        exe = find_executable('armcc')
```

Comment:
Is there any way to move this inside the `not TOOLCHAIN_PATHS['ARM']` check? If we already have a path that's set, there's no need to try and find the executable.

This applies to `GCC_ARM` and `IAR` as well.